### DISCUSSION TITLE
Release Nvidia Jetson Standard Kairos Image

### DISCUSSION THREAD

**Discussion by laurivosandi** [kairos-io/kairos#2922] [Polls]
This feature is up for voting 🗳️, before we release standard artifacts for the Nvidia Jetson bords, we'd like to understand how many users would be interested in these images. This is because we have limited CI resources and the team doesn't have enough boards to test changes.

**Is your feature request related to a problem? Please describe.**

No k3s images are available for nvidia-jetson-agx-orin

**Describe the solution you'd like**

I'd like to be able to use k3s on Orin boards

**Describe alternatives you've considered**

Tried to locally build with:

```
earthly -P +all-arm \
  --VARIANT=standard \
  --MODEL=nvidia-jetson-agx-orin \
  --FLAVOR=ubuntu \
  --FLAVOR_RELEASE=20.04 \
  --FAMILY=ubuntu \
  --BASE_IMAGE=quay.io/kairos/cache:nvidia-base \
  --IMG_COMPRESSION=false \
  --K3S_VERSION=1.31.1
```

It succeeded but it seems there is no way to locally use the built image with `osbuilder-tools` or `luet`?
![Screenshot from 2024-09-27 10-52-10](https://github.com/user-attachments/assets/86922a7b-258f-42da-8b78-84560b6f7dc8)

**Additional context**

**Comment by mudler** [comment]
@laurivosandi for Nvidia AGX we don't publish yet standard images with k3s.

For the moment it would be enough to use the images currently published (without k3s) and overlay on top the k3s/provider packages.

For instance, from our Dockerfiles, this is what standard images are including:
```
FROM base-kairos AS kairos-standard
ARG SOFTWARE_VERSION
LABEL io.kairos.k3s_version="${SOFTWARE_VERSION}"
RUN luet install -y system/provider-kairos
RUN luet install -y "k8s/k3s-$(which-init.sh)@${SOFTWARE_LUET_VERSION:-$SOFTWARE_VERSION}" utils/edgevpn utils/k9s utils/nerdctl container/kubectl utils/kube-vip
```

You can install k3s and provider-kairos with luet, or just overlay the container images with `COPY / / --from=container-image`

**Comment by mauromorales** [comment]
@laurivosandi I've switched this to a poll because:

> we'd like to understand how many users would be interested in these images. This is because we have limited CI resources and the team doesn't have enough boards to test changes.

For the resulting image that is built, you need to extract the `img` image inside the container image, e.g.:

```
export IMAGE=quay.io/kairos/opensuse:tumbleweed-core-arm64-rpi4-v3.2.1-img
docker run -ti --rm -v $PWD:/image quay.io/luet/base util unpack "$IMAGE" /image
```

Let me know if that works

**Reply by jbalonso** [reply]
Strictly speaking, I have an old Nvidia Jetson TX2 that's not likely to work with modern anything, but I've recently dusted it off to see if I can add it to my k3s cluster.
